liblouis/lou_translateString.c

```diff
@@ -535,13 +541,15 @@ doPassSearch(const TranslationTableHeader *table, InString input,
 				for (k = 0; k < passInstructions[*searchIC + 3]; k++) {
 					if (input.chars[*searchPos] == ENDSEGMENT)
 						itsTrue = 0;
-					else
+					else {
 						itsTrue = ((findCharOrDots(input.chars[(*searchPos)++],
 											passCharDots,
 											table)->attributes &
 										   attributes)
 										? 1
 										: 0);
+						if (not) itsTrue = !itsTrue;
```

Comment:
This code is so awful it hurts, but I guess that is how it was before. 
``` C
not = 0; // not what?
itsTrue = !itsTrue // WAT? Is it not true or not? And what is true?
if (!not) break;// OMG
```

Reply:
`itsTrue` means it's a match, `not` is true when there was a exclamation mark. I agree it's awful but it was like that and I don't want to refactor the whole function now.

Reply:
Hehe, totally understand and agree. Maybe just add a comment to the vars with above explanation

Reply:
But that explanation hardly helps anything. Just some papering over the cracks. It's the code itself that sucks :(

Reply:
> papering over the cracks

funny analogy, I hope it does help somewhat

Reply:
Is it not an English expression? I look it up in a dictionary :)

Reply:
Oh, I don't know. I though it was maybe some Flemish thing

Reply:
No, we would say "doekje voor het bloeden".